### DISCUSSION TITLE
Welcome to metatrader5-linux-django-docker Discussions!

### DISCUSSION THREAD

**Discussion by slowfound** [sesto-dev/metatrader5-quant-server-python#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chonguoimuon** [comment]
How to send a request to open/close a position to a connected forex account?

**Comment by chonguoimuon** [comment]
I have access to apidocs/#/ but when I create a new order I always get the response "Invalid order type"

**Reply by slowfound** [reply]
Depending on your broker and the pair you're trying to trade, there are different filling types available to you. Take a look at this:

https://www.mql5.com/en/docs/constants/tradingconstants/orderproperties#enum_order_type_filling

I've used `ORDER_FILLING_FOK` [here on line 97 in the entry.py](https://github.com/sesto-dev/metatrader5-quant-server-python/blob/main/backend/django/app/quant/algorithms/mean_reversion/entry.py) you can change it to other filling types to see if that fixes the problem for you.

**Reply by chonguoimuon** [reply]
> Depending on your broker and the pair you're trying to trade, there are different filling types available to you. Take a look at this:
> 
> https://www.mql5.com/en/docs/constants/tradingconstants/orderproperties#enum_order_type_filling
> 
> I've used `ORDER_FILLING_FOK` [here on line 97 in the entry.py](https://github.com/sesto-dev/metatrader5-quant-server-python/blob/main/backend/django/app/quant/algorithms/mean_reversion/entry.py) you can change it to other filling types to see if that fixes the problem for you.

i am rest api in swagger of chapter 2. currently my need is simple just open position and close them through api. but python app is not working.
when open position will get error : "Error in send_market_order: 'NoneType' object has no attribute 'retcode'"
i sent request as below :"{'action': 1, 'symbol': 'BTCUSDm', 'volume': 0.01, 'type': 0, 'price': 91128.71, 'deviation': 0, 'magic': 0, 'comment': 'test', 'type_time': 0, 'type_filling': 'ORDER_FILLING_IOC'}"
when close all return error :" Failed to close position 1215470645: Invalid request"
can you guide me to edit python files in app?

**Reply by chonguoimuon** [reply]
I found a bug in chapter-2 fixed it and tested it on demo account. patches are suggested

**Comment by Shak-Shat** [comment]
when is chapter 5 releasing ?

**Comment by FelixFlxn** [comment]
Hey man, when i try to install it the way its described in the video (chapter1), it says, my mt5 version is deprecated :( what do i have to change? is it the version number in script 02-common? i cant seem to find any version numbers of metatrader5. i also ound a error message in the logs: Cannot load image "/config/.wine/drive_c/Program Files/MetaTrader 5/Terminal.ico" from file "/config/.wine/drive_c/Program Files/MetaTrader 5/Terminal.ico"
